base/libgit2/merge.jl

```diff
@@ -1,5 +1,18 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
+"""
+    GitAnnotated(repo::GitRepo, commit_id::GitHash)
+    GitAnnotated(repo::GitRepo, ref::GitReference)
+    GitAnnotated(repo::GitRepo, fh::FetchHead)
+    GitAnnotated(repo::GitRepo, comittish::AbstractString)
+
+An annotated git commit carries with it information about how it was looked up and
+why, so that rebase or merge operations have more information about the context of
+the commit. Conflict files contain information about the source/target branches in
+the merge which are conflicting, for instance. An annotated commit can refer to the
+tip of a remote branch, for instance when a [`FetchHead`](@ref) is passed, or to a
+branch head described using `GitReference`.
```

Comment:
I really like this description! It would be good to mention what the `GitAnnotated` constructor does or returns though. It gives you a `GitAnnotated`, which is an annotated commit object, right?

Reply:
Yeah, I guess it's fine as-is.